vulnfeeds/cves/versions.go

```diff
@@ -71,30 +141,55 @@ func ExtractVersionInfo(cve CVEItem, validVersions []string) VersionInfo {
 		}
 	}
 
+	gotVersions := false
 	for _, node := range cve.Configurations.Nodes {
 		if node.Operator != "OR" {
 			continue
 		}
 
-		// TODO: Also try to parse description as these are not always reliably set.
 		for _, match := range node.CPEMatch {
 			if !match.Vulnerable {
 				continue
 			}
 
-			if match.VersionStartIncluding != "" || match.VersionEndExcluding != "" {
-				if match.VersionStartExcluding != "" || match.VersionEndIncluding != "" {
-					// TODO: handle these by using validVersions.
-					continue
-				}
-
-				v.AffectedVersions = append(v.AffectedVersions, AffectedVersion{
-					// TODO: make sure these match validVersions.
-					Introduced: match.VersionStartIncluding,
-					Fixed:      match.VersionEndExcluding,
-				})
+			introduced := ""
+			fixed := ""
+			if match.VersionStartIncluding != "" {
+				introduced = match.VersionStartIncluding
+			} else if match.VersionStartExcluding != "" {
+				introduced = nextVersion(validVersions, match.VersionStartExcluding)
+			}
+
+			if match.VersionEndExcluding != "" {
+				fixed = match.VersionEndExcluding
+			} else if match.VersionEndIncluding != "" {
+				fixed = nextVersion(validVersions, match.VersionEndIncluding)
 			}
+
+			if introduced == "" && fixed == "" {
+				continue
+			}
+
+			if introduced != "" && !hasVersion(validVersions, introduced) {
+				log.Printf("Warning: %s is not a valid introduced version", introduced)
```

Comment:
in these cases should we still consider that we "Got the version"?

Reply:
Yes, there will always still be a human element involved. This will flag cases in the future that require a set of eyes to review.